sablier-labs/v2-periphery
# feat: add `name`, introduces struct `ConstructorParams`

closes #233 

Adding `name` throws `stack too deep` error because of the large number of variables in function `createMerkleStreamerLL`. In order to fix that, this PR introduces a new struct `ConstructorParams` for `SablierV2MerkleStreamer` constructor parameters.

```solidity
struct ConstructorParams {
    address initialAdmin;
    IERC20 asset;
    string name;
    bytes32 merkleRoot;
    uint40 expiration;
    bool cancelable;
    bool transferable;
}
```

## Summary

### Source
- uses struct `ConstructorParams` to encapsulate `SablierV2MerkleStreamer` constructor parameters.
- adds `name` parameter of type string.
- adds a new internal storage `NAME` of type `bytes32` to store campaign name. `bytes32` is preferred over `string` to optimize for gas.
- adds a public function `name()` to retrieves the campaign name as `string`.

### Events
- update event `CreateMerkleStreamerLL`.
- restructures the order of parameters.

### Errors
- adds a new error `SablierV2MerkleStreamer_CampaignNameTooLong`.

### Tests
- defines `baseParams ` to retrieve default value of `ConstructorParams` struct.
- add `test_RevertWhen_CampaignNameTooLong` to check for `name` length
- update `createMerkleStreamerLL.tree`
